### DISCUSSION TITLE
Is it possible to generate a bubbletea "form" from JsonSchema?

### DISCUSSION THREAD

**Discussion by iekdosha** [charmbracelet/bubbletea#342] [Ideas]
I am looking for an automatic way to create a bubbletea application from a json schema file.

Is that or something similar supported?

Thanks :)

**Comment by bashbunni** [comment]
Hi @iekdosha are you asking if you can create a full Bubble Tea application from a JSON schema file? Or are you asking if you can create an input form in Bubble Tea with JSON schema file? Either way, what would you expect that file look like? 

To the best of my knowledge, this isn't something we support at the moment. We would be interested in hearing more information about your proposal though!

**Comment by chavacava** [comment]
Hi, I know this thread is a bit old but I found it while searching how to generate TUI from JSON Schema.
In the world of web interfaces there are some libraries able to generate a web form from a JSON schema (for example [this one](https://rjsf-team.github.io/react-jsonschema-form/) and [this another one](https://jsonforms.io/))

I'm looking something equivalent for developing a TUI thus I've instantly thought on bubbletea :) 
AFAIK there is no such a functionality in bubbletea... I think it will be nice to consider adding it.

**Reply by bashbunni** [reply]
Hey @chavacava [Huh](github.com/charmbracelet/huh) is a form-specific wrapper of Bubble Tea that might be better suited to your goals here. If you were looking to generate forms from some kind of template, I think Huh's API would be easier to work with
